src/getDataTransferItems.js

```diff
@@ -1,17 +1,89 @@
-export default function getDataTransferFiles(event) {
+function traverseDirectory(entry, path, entriesList) {
+  const reader = entry.createReader()
+  // Resolved when the entire directory is traversed
+  return new Promise(resolveDirectory => {
+    const iterationAttempts = []
+    ;(function readEntries() {
+      // According to the FileSystem API spec, readEntries() must be called until
+      // it calls the callback with an empty array.  Seriously??
+      reader.readEntries(entries => {
+        if (!entries.length) {
+          // Done iterating this particular directory
+          resolveDirectory(Promise.all(iterationAttempts))
+        } else {
+          // Add a list of promises for each directory entry.  If the entry is itself
+          // a directory, then that promise won't resolve until it is fully traversed.
+          iterationAttempts.push(
+            Promise.all(
+              entries.map(entryObj => {
+                if (entryObj.isFile) {
+                  return new Promise(resolve => {
+                    entryObj.file(file => {
+                      file.fullPath = `${path}/${file.name}` // eslint-disable-line no-param-reassign
```

Comment:
I attempted something similar in html5-file-selector (manually setting the `file.fullPath` to the actual nested path), but ran into errors with immutability of File objects with one browser - can't remember which one.  So I ended up wrapping the File object in a wrapper so that this data could be reliably passed back to the client code.